libs/datasets/sources/cdc_testing_dataset.py

```diff
@@ -58,7 +58,60 @@ class CDCHistoricalTestingDataset(data_source.CanScraperBase):
     @classmethod
     @lru_cache(None)
     def make_dataset(cls) -> MultiRegionDataset:
-        return modify_dataset(super().make_dataset())
+        ds = super().make_dataset()
+        return dataclasses.replace(
+            ds, timeseries=remove_trailing_zeros(ds.timeseries), timeseries_bucketed=None
+        )
+
+
+class CDCOriginallyPostedTestingDataset(CDCHistoricalTestingDataset):
+    """Data source connecting to the official as-originally-posted CDC test positivity dataset."""
+
+    SOURCE_TYPE = "CDCOriginallyPostedTesting"
+
+    VARIABLES = [
+        ccd_helpers.ScraperVariable(
+            variable_name="pcr_tests_positive",
+            measurement="rolling_average_7_day",
+            provider="cdc_originally_posted",
+            unit="percentage",
+            common_field=CommonFields.TEST_POSITIVITY_7D,
+        ),
+    ]
+
+
+class CDCCombinedTestingDataset(data_source.DataSource):
+    """Data source combining the CDC's historical and as-originally-posted datasets."""
+
+    # This gets overwritten by the tags generated in CDCOriginallyPostedTestingDataset.
+    # We use the as-originally-posted source for the tags as this is the dataset with the
+    # most up-to-date data points
+    SOURCE_TYPE = "CDCTesting"
+    EXPECTED_FIELDS = [CommonFields.TEST_POSITIVITY_7D]
+
+    @classmethod
+    @lru_cache(None)
+    def make_dataset(cls) -> MultiRegionDataset:
+        """Use historical data when possible, use as-originally-posted for data that the historical dataset does not have.
+        
+        The historical data is usually delayed behind the as-originally-posted by ~3 days. 
+        To get the combination of the most up-to-date and accurate data we use the historical dataset
+        by default, but use as-originally-posted for the most recent days that the historical dataset
+        does not yet have.
+        """
+
+        historical_ds = CDCHistoricalTestingDataset().make_dataset()
+        as_posted_ds = CDCOriginallyPostedTestingDataset().make_dataset()
+
+        historical_ts = historical_ds.timeseries
+        as_posted_ts = as_posted_ds.timeseries
+
+        merged_ts = historical_ts.combine_first(as_posted_ts)
+
+        merged_ds = dataclasses.replace(
+            as_posted_ds, timeseries=merged_ts, timeseries_bucketed=None,
+        )
+        return modify_dataset(merged_ds)
```

Comment:
Can we call `modify_dataset()` directly in `CDCOriginallyPostedTestingDataset` and `CDCHistoricalTestingDataset` so that they behave as "valid" data sources? (e.g. with test positivity as a ratio instead of percent).  Then if we needed to we could safely switch to `CDCHistoricalTestingDataset` as our source of choice in the future without having to modify this code.

Reply:
Smart, I agree it's best to have each piece be as complete as possible, updated to follow this in 2f849b8c4b9051e8cef29ee55fcb650dfa179a1c